src/embeddings.rs

```diff
@@ -269,6 +269,27 @@ impl PyEmbeddings {
     }
 }
 
+#[pyproto]
+impl PyMappingProtocol for PyEmbeddings {
+    fn __getitem__(&self, word: &str) -> PyResult<Py<PyArray1<f32>>> {
+        let embeddings = self.embeddings.borrow();
+
+        use EmbeddingsWrap::*;
```

Comment:
Since we now look up embeddings in two places, this could be factored out to a method in `EmbeddingsWrap`.